src/services/breakpoints.js

```diff
@@ -1,21 +1,70 @@
-const BREAKPOINT_TABLET = '(min-width:768px)';
-const BREAKPOINT_DESKTOP = '(min-width:1280px)';
+import onResize from 'hd-blocks/services/on-resize';
 
-let breakpoints = {
-  tablet: BREAKPOINT_TABLET,
-  desktop: BREAKPOINT_DESKTOP,
+let breakpoints = {};
+const matchMediaAvailable = (
+  typeof window !== 'undefined'
+  && typeof window.matchMedia === 'function'
+);
+
+export const mediaMatches = (breakpoint, {
+  matchMedia = matchMediaAvailable ? window.matchMedia : () => false,
+} = {}) => {
+  if (typeof breakpoints[breakpoint] === 'undefined') {
+    // Breakpoint not defined, we treat it as a media query string
+    return matchMedia(breakpoint).matches;
+  }
+
+  return matchMedia(breakpoints[breakpoint]).matches;
 };
 
-export const isMobile = () => !window.matchMedia(breakpoints.tablet).matches;
-export const isDesktop = () => window.matchMedia(breakpoints.desktop).matches;
-export const isTablet = () => !isDesktop() && window.matchMedia(breakpoints.tablet).matches;
 export const setBreakpoints = (newBreakpoints) => {
-  breakpoints = { ...breakpoints, ...newBreakpoints };
+  breakpoints = newBreakpoints;
 };
 
-export default {
-  isMobile,
-  isTablet,
-  isDesktop,
-  setBreakpoints,
+export const getBreakpoints = () => breakpoints;
```

Comment:
It looks a bit strange, having component inside of service here. And this component is also persisting state (breakpoints. right?). How about decoupling setting and getting breakpoints and keep it in `breakpoints` service and taking the component out of there. Then we'll have `Responsive` component, which could take also `breakpoints` object, if not passed, it would check the service, if there are global breakpoints? 
So you would also be able to pass specific breakpoints to the responsive component, instead of overwriting them on global level, if you need on the fly modification. Then I think the split is also clearer, where breakpoints service is responsible for setting and updating global breakpoints and just that. And Responsive component is responsible of consuming the breakpoints and falling back.

Reply:
Thanks for a great idea, that's a good suggestion. Will implement it.

Reply:
Done :)